content/src/controller/Controller.ts

```diff
@@ -251,26 +253,55 @@ export class Controller {
         // Method: GET
         // Path: /history
         // Query String: ?from={timestamp}&to={timestamp}&serverName={string}
-        const from       = req.query.from
-        const to         = req.query.to
-        const serverName = req.query.serverName
-        const offset     = this.asInt(req.query.offset)
-        const limit      = this.asInt(req.query.limit)
-
-        const history = await this.service.getLegacyHistory(from, to, serverName, offset, limit)
-        res.send(history)
+        const fromOriginTimestamp = req.query.from
+        const toOriginTimestamp   = req.query.to
+        const serverName          = req.query.serverName
+        const offset              = this.asInt(req.query.offset)
+        const limit               = this.asInt(req.query.limit)
+
+        let originServerUrl: ServerAddress | undefined
+        if (serverName) {
+            originServerUrl = this.cluster.getAddressForServerName(serverName) ?? 'UNKNOWN_NAME'
+        }
```

Comment:
```suggestion
        const originServerUrl: ServerAddress | undefined = serverName ? decodeURIComponent(serverName) : undefined
```

Also, I think we can delete the method `getAddressForServerName`